crates/nu-explore/src/views/record/mod.rs

```diff
@@ -23,25 +23,39 @@ use nu_protocol::{
     Config, Record, Span, Value,
 };
 use ratatui::{layout::Rect, widgets::Block};
-use std::{borrow::Cow, collections::HashMap};
+use std::collections::HashMap;
 
 pub use self::table_widget::Orientation;
 
 #[derive(Debug, Clone)]
-pub struct RecordView<'a> {
-    layer_stack: Vec<RecordLayer<'a>>,
+pub struct RecordView {
+    layer_stack: Vec<LayerData>,
     mode: UIMode,
     orientation: Orientation,
     cfg: ExploreConfig,
 }
 
-impl<'a> RecordView<'a> {
-    pub fn new(
-        columns: impl Into<Cow<'a, [String]>>,
-        records: impl Into<Cow<'a, [Vec<Value>]>>,
-    ) -> Self {
+#[derive(Debug, Clone)]
+struct LayerData {
+    record: RecordLayer,
+    widget_data: Option<RecordData>,
+}
+
+#[derive(Debug, Clone)]
+struct RecordData {
+    columns: Vec<String>,
+    records: Vec<Vec<NuText>>,
```

Comment:
Would it make sense for this data to live inside of `RecordLayer`?

We have 3 separate structs now: `LayerData`, `RecordLayer`,and `RecordData`. I'm not sure we need all 3. What if we do this?

1. Delete `LayerData` and `RecordData`
2. Add a new `records_text: Vec<Vec<NuText>>` field to `RecordLayer`, and set it in `RecordLayer::new()`

I'm happy to make this change, just wanted to check with you first.

Reply:
> We have 3 separate structs now: LayerData, RecordLayer,and RecordData. I'm not sure we need all 3

Yeeeee also have smells about it.

I guess it'll be OK